web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -553,6 +553,22 @@ export type NodeMeta = BaseMeta & {
   node: Node;
 };
 
+export type DatabaseServiceDeploy = {
+  /**
+   * serviceDeployedMethod flag will be undefined if user skipped
+   * deploying service (service already existed).
+   */
+  method: ServiceDeployMethod;
+  /**
+   * this field is only defined for auto deployed services,
+   */
+  selectedSecurityGroups?: string[];
+  /**
+   * this field is only defined for auto deployed services
+   */
+  selectedSubnetIds?: string[];
+};
```

Comment:
If you want, you can make this type more strict (this will require some adjustments on the callsites).
```suggestion
export type DatabaseServiceDeploy =
  | {
      method: 'auto';
      selectedSecurityGroups: string[];
      selectedSubnetIds: string[];
    }
  | {
      method: 'manual' | 'skipped';
    };
```